src/layered_config_tree/main.py

```diff
@@ -231,8 +232,8 @@ class LayeredConfigTree:
 
     def __init__(
         self,
-        data: Union[Dict, str, Path, "LayeredConfigTree"] = None,
-        layers: List[str] = None,
+        data: Optional[Union[Dict[str, Any], str, Path, "LayeredConfigTree"]] = None,
+        layers: Optional[List[str]] = None,
```

Comment:
To be honest, I'm pretty confused by layers, source, and data and whther they are allowed to be empty or None. It might be worth just talking through a few of these comments